cmd/upgrade.go

```diff
@@ -0,0 +1,64 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/railwayapp/cli/constants"
+	"github.com/railwayapp/cli/entity"
+)
+
+var OriginalInstallationMethod string
+
+func getExectuablePath() string {
+	ex, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+	exPath := filepath.Dir(ex)
+	return exPath
+}
+
+func getUpdateCommand(installationMethod string) (*exec.Cmd, error) {
+	if installationMethod == "brew" {
+		return exec.Command("brew", "upgrade", "railway"), nil
+	} else if installationMethod == "curl" {
+		return exec.Command("curl", "-fsSL", "https://railway.app/install.sh", "|", "sh"), nil
+	} else if installationMethod == "npm" {
+		currentPath := getExectuablePath()
+		if strings.Contains(currentPath, "npm") {
+			return exec.Command("npm", "i", "-g", "@railway/cli"), nil
+		} else if strings.Contains(currentPath, "yarn") {
+			return exec.Command("yarn", "global", "add", "@railway/cli"), nil
+		}
```

Comment:
I'm actually not sure how you'd distribute multiple binaries for each of these :/

Frankly, you may want to just go with the `getExecutablePath()` method for homebrew/CLI, and then if you can't detect the installation for upgrade throw